### PR TITLE
Feature: add unified on CRI to support cgroup v2

### PR DESCRIPTION
**Pull request by payall4u**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This CRI update is intended to support more cgroupv2 capabilities.

#### Does this PR introduce a user-facing change?

```release-note
Add unified map on CRI to support cgroup v2. Refer to https://github.com/opencontainers/runtime-spec/blob/master/config-linux.md#unified.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

- [KEP]: [Support memory qos with cgroups v2](https://github.com/kubernetes/enhancements/pull/2571)

### CODE REVIEW

**Comment by pacoxu** [comment]
/cc xiaoxubeii 
https://github.com/kubernetes/enhancements/issues/2570

**Comment by payall4u** [comment]
> Thanks for your pull request. Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please follow instructions at https://git.k8s.io/community/CLA.md#the-contributor-license-agreement to sign the CLA.**
> 
> It may take a couple minutes for the CLA signature to be fully registered; after that, please reply here with a new comment and we'll verify. Thanks.
> 
> * If you've already signed a CLA, it's possible we don't have your GitHub username or you're using a different email address.  Check your existing CLA data and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * If you signed the CLA as a corporation, please sign in with your organization's credentials at https://identity.linuxfoundation.org/projects/cncf to be authorized.
> * If you have done the above and are still having issues with the CLA being reported as unsigned, please log a ticket with the Linux Foundation Helpdesk: https://support.linuxfoundation.org/
> * Should you encounter any issues with the Linux Foundation Helpdesk, send a message to the backup e-mail support address at: [login-issues@jira.linuxfoundation.org](mailto:login-issues@jira.linuxfoundation.org)

@k8s-ci-robot ping

**Comment by pacoxu** [comment]
Should `staging/src/k8s.io/api/core/v1/generated.proto` be updated as well?

**Comment by payall4u** [comment]
> Should `staging/src/k8s.io/api/core/v1/generated.proto` be updated as well?

We needn't update it now, I think.

File: staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto [hunk 1]
```diff
@@ -585,6 +585,8 @@ message LinuxContainerResources {
     string cpuset_mems = 7;
     // List of HugepageLimits to limit the HugeTLB usage of container per page size. Default: nil (not specified).
     repeated HugepageLimit hugepage_limits = 8;
+    // Unified resources for cgroup v2. Default: nil (not specified).
+    map<string, string> unified = 9;
```
Comment (AkihiroSuda) [inline staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto:589]: Could you clarify the string format

Comment (payall4u) [reply staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto:590]: Done

Comment (payall4u) [reply staging/src/k8s.io/cri-api/pkg/apis/runtime/v1alpha2/api.proto:590]: > Could you clarify the string format

Done

**Review by ehashman — COMMENTED** [review]
/triage accepted
/priority important-soon

Are there e2e tests for this? Won't this need a release note?

**Comment by payall4u** [comment]
> Won't this need a release note?

Sorry. I had just finished it.

> Are there e2e tests for this? 

I think there is no such e2e test for cri before. We can add the test with [Support memory qos with cgroups v2](https://github.com/kubernetes/enhancements/pull/2571) together.

**Review by xiaoxubeii — APPROVED** [review]

**Comment by payall4u** [comment]
ping @AkihiroSuda @mrunalp

**Comment by ehashman** [comment]
Note: there is a conflicting CRI protobuf change in https://github.com/kubernetes/kubernetes/pull/102823/files#diff-7a3c2723dedc6b44cf183fc52f84cc4015ffbd65a998a65c3959a953cb63c945

**Review by mrunalp — CHANGES_REQUESTED** [review]
Could you also add it to the v1/api.proto file?

**Comment by payall4u** [comment]
> Could you also add it to the v1/api.proto file?

Done.

**Review by mrunalp — APPROVED** [review]

**Pull request merged**